### DISCUSSION TITLE
Idea: dual battery support

### DISCUSSION THREAD

**Discussion by bjwil1991** [valnoxy/t480-oc#34] [Ideas]
Here's an idea: supporting 2 batteries for the T480 running either Monterey or Ventura (beta) or even combining both of them into 1 battery for longer battery life (hours until they're empty).

The reason for that is to see how much time the batteries will have until they run out and the OS only detects the internal and not the external (probably a kext requirement, but I digress).

**Comment by WigHlod** [comment]
Hello, i've read on somes Reddit post that the SMCBatteryManager.kext already merging the two battery together and show them as a big one. 

Have them managed separately would be great but look not so easy to achieve.

**Reply by trevorwdunn** [reply]
Can confirm this is the case. Go into System Information and look at the Wh listed there for your battery. Should be the sum of the internal and external batteries.

**Comment by arvinsingla** [comment]
Reviving this old discussion. I'm running 1.3.0 with Ventura and am not seeing the sum of both batteries together. In fact I'm not even seeing the total for one of the batteries. I'm still relatively new to this so unsure 

Here I see 1406 mAh when it should be significantly higher 
![Screenshot 2023-09-26 at 8 30 26 AM 1](https://github.com/valnoxy/t480-oc/assets/645082/717de786-0640-4c5a-a410-a39b6957665b)

Using IORegistryExplorer seems to only show a single battery
![Screenshot 2023-09-26 at 8 37 03 AM](https://github.com/valnoxy/t480-oc/assets/645082/325a3f1f-5662-4306-80cc-403f8db093f6)

The OpenCore install guide talks about [merging the two batteries](https://dortania.github.io/OpenCore-Post-Install/laptop-specific/battery.html#battery-hot-patching) using the [guide here](https://github.com/acidanthera/VirtualSMC/blob/master/Docs/Dual%20Battery%20Support.md) but this is a bit over my head.

Does anyone have thoughts or tips on how to get both batteries combined and working together?

**Reply by trevorwdunn** [reply]
What is the combined wattage between both of your batteries?

**Reply by arvinsingla** [reply]
The combined capacity should be 3900 mAh. The internal and external battery are each 1950mAh.

**Reply by arvinsingla** [reply]
Upon further investigation I believe this is an error on my part. I booted into an ubuntu USB key and ran acpi battery details and it turns out the battery was not making a proper connection and wasn't beign detected. I removed and re-attached it and it seems to be working correctly now. It also provided details on why the individual number was low. Even though the design capacity was 1950 the current full capacity has degraded.

**Reply by trevorwdunn** [reply]
Glad it worked out. You may want to check your prongs.
